node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -75,58 +78,114 @@ case class P2PClientActor(
     * This is the context we are in after we received a [[Tcp.Connected]] message
     */
   private def awaitNetworkRequest(
-      peer: ActorRef,
+      peerConnection: ActorRef,
       unalignedBytes: ByteVector): Receive =
     LoggingReceive {
       case message: NetworkMessage =>
-        sendNetworkMessage(message, peer)
+        sendNetworkMessage(message, peerConnection)
       case payload: NetworkPayload =>
         val networkMsg = NetworkMessage(network, payload)
         self.forward(networkMsg)
       case message: Tcp.Message =>
         val newUnalignedBytes =
-          handleTcpMessage(message, Some(peer), unalignedBytes)
-        context.become(awaitNetworkRequest(peer, newUnalignedBytes))
+          handleTcpMessage(message, peerConnection, unalignedBytes)
+        context.become(awaitNetworkRequest(peerConnection, newUnalignedBytes))
       case metaMsg: P2PClient.MetaMsg =>
         sender() ! handleMetaMsg(metaMsg)
+      case Terminated(actor) if actor == peerConnection =>
+        context stop self
     }
 
-  /** This context is responsible for initializing a tcp connection with a peer on the bitcoin p2p network */
-  def receive: Receive =
-    LoggingReceive {
-      case cmd: Tcp.Command =>
-        //we only accept a Tcp.Connect/Tcp.Connected
-        //message to the default receive on this actor
-        //after receiving Tcp.Connected we switch to the
-        //'awaitNetworkRequest' context. This is the main
-        //execution loop for the Client actor
-        handleCommand(cmd, peerOpt = None)
-
-      case connected: Tcp.Connected =>
-        val _ = handleEvent(connected, unalignedBytes = ByteVector.empty)
-      case msg: NetworkMessage =>
-        self.forward(msg.payload)
-      case payload: NetworkPayload =>
-        logger.error(
-          s"Cannot send a message to our peer when we are not connected! payload=${payload} peer=${peer}")
+  def receive: Receive = {
+    case P2PClient.ConnectCommand =>
```

Comment:
Are you intentionally getting rid of the `LoggingReceive` directive? It's useful for debugging. 

https://doc.akka.io/docs/akka/current/logging.html#auxiliary-logging-options

This might be useful for both @shreyanshyad and @shruthii625 to see each individual message that is passed to the actor.